File: packages/java/src/main/java/com/ummalqura/ValidationResult.java
package com.ummalqura;

/**
 * Validation result
 */
public class ValidationResult {
    private final boolean valid;
    private final CalendarError error;

    public ValidationResult(boolean valid, CalendarError error) {
        this.valid = valid;
        this.error = error;
    }

    public boolean isValid() { return valid; }
    public CalendarError getError() { return error; }
}

File: packages/php/src/ValidationResult.php
<?php

namespace UmmAlQura;

class ValidationResult
{
    public function __construct(
        private bool $valid,
        private ?CalendarError $error = null
    ) {}

    public function isValid(): bool { return $this->valid; }
    public function getError(): ?CalendarError { return $this->error; }

    public function toArray(): array
    {
        $result = ['valid' => $this->valid];
        if ($this->error !== null) {
            $result['error'] = $this->error->toJson();
        }
        return $result;
    }
}
